ngraph/core/reference/include/ngraph/runtime/reference/helpers.hpp

```diff
@@ -0,0 +1,44 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            struct widen
+            {
+                using type = T;
+            };
+
+            template <>
+            struct widen<float>
+            {
+                using type = double;
+            };
+
+            template <>
+            struct widen<double>
+            {
+                using type = long double;
+            };
```

Comment:
@lazarevevgeny @ilyachur could you comment on this widening technique used here ? The previous implementation used wider type for accumulation than the the output type, so I left it as it was before. The same technique is used in dot operator. This is not really specified anywhere and we can have different results in implementations if it is not followed uniformly.  Should we always do computation in widest possible type also in other operators?  Is this approach consistent across the plugins?

Reply:
Can we use original data types or do you see any issues?

Reply:
I will try to use original type and we will see if SLT finds any issues with precision regarding to plugins. 

Another topic. We agreed with Evgeny that we need the same type for input and kernels so it implies also the same type of output. Now, if we don't need widening we can reduce type parameters of convolution() to 1 only.

`convolution<INPUT, FILTER, OUTPUT, ACCU>()` ->`convolution<T>()`

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/3922/commits/f0e720b5bbec8a4c94b040e95c67386cc4522efd). No issues came out.